WordPress/src/main/java/org/wordpress/android/ui/uploads/LocalDraftUploadStarter.kt

```diff
@@ -0,0 +1,37 @@
+package org.wordpress.android.ui.uploads
+
+import android.content.Context
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.store.PostStore
+import org.wordpress.android.modules.BG_THREAD
+import org.wordpress.android.util.NetworkUtilsWrapper
+import javax.inject.Inject
+import javax.inject.Named
+
+/**
+ * Provides a way to find and upload all local drafts.
+ */
+class LocalDraftUploadStarter @Inject constructor(
+    /**
+     * The Application context
+     */
+    private val context: Context,
+    private val postStore: PostStore,
+    /**
+     * The Coroutine dispatcher used for querying in FluxC.
+     */
+    @Named(BG_THREAD) private val bgDispatcher: CoroutineDispatcher,
+    private val networkUtilsWrapper: NetworkUtilsWrapper
+) {
+    fun uploadLocalDrafts(scope: CoroutineScope, site: SiteModel) = scope.launch(bgDispatcher) {
+        if (!networkUtilsWrapper.isNetworkAvailable()) {
+            return@launch
+        }
+
+        val localDrafts = postStore.getLocalDrafts(site)
+        localDrafts.forEach { UploadService.uploadPost(context, it) }
```

Comment:
I might have misread `UploadService.unpackPostIntent`. I saw the usage of `PostUploadHandler.isPostUploadingOrQueued` in there and just assumed that it will ignore posts that are currently in the queue. I should have tested that. 

My original plan for this class was for it to be a `@Singleton` and manage its own queue so it will not do a query and upload in a short timeframe. I abandoned that to keep it simple and get your feedback. 🙂 Which turned out to be a good decision. I'll wait for the discussion with Maxime and everyone else before moving forward.

Reply:
I ended up just using `UploadService.isPostUploadingOrQueued` in fa94f00d0d9c015719bd9e3b6ad50c062396eb15. I figured we'll get to the complicated queuing portion in #9846.